src/core/server/opensearch/client/configure_client.ts

```diff
@@ -30,25 +30,37 @@
 
 import { Buffer } from 'buffer';
 import { stringify } from 'querystring';
-import { Client } from '@opensearch-project/opensearch';
+import { Client, ClientOptions } from '@opensearch-project/opensearch';
+import {
+  Client as ClientNext,
+  ClientOptions as ClientOptionsNext,
+} from '@opensearch-project/opensearch-next';
 import { RequestBody } from '@opensearch-project/opensearch/lib/Transport';
 
 import { Logger } from '../../logging';
 import { parseClientOptions, OpenSearchClientConfig } from './client_config';
 
-export const configureClient = (
+type ClientType<T> = T extends { withLongNumeralsSupport: true } ? ClientNext : Client;
+
+export const configureClient = <T>(
   config: OpenSearchClientConfig,
-  { logger, scoped = false }: { logger: Logger; scoped?: boolean }
-): Client => {
+  {
+    logger,
+    scoped = false,
+    withLongNumeralsSupport = false,
+  }: T extends { logger: Logger; scoped?: boolean; withLongNumeralsSupport?: boolean } ? T : never
+): ClientType<T> => {
   const clientOptions = parseClientOptions(config, scoped);
 
-  const client = new Client(clientOptions);
+  const client = withLongNumeralsSupport
+    ? new ClientNext({ ...clientOptions, enableLongNumeralSupport: true } as ClientOptionsNext)
+    : new Client(clientOptions as ClientOptions);
```

Comment:
Why do we need to avoid `ClientNext` by default? Is there a breaking change failure case that's supported but incompatible with long numeral support?

Reply:
The simplest breaking change is introduced by differing typing as a result of changes to `ClientOptions`. It is the same reason that we couldn't replace the JS client v1 with v2 in 2.x earlier.